Add components

jnation3406 opened the pull request.

Commit:
components pages initial commit

Commit:
a few small tweaks in the components section

jnation3406 commented on _config.yml:
```diff
@@ -1,4 +1,3 @@
 # Specify remote theme so github pages can load it from the author's repository
 remote_theme: chrisrhymes/bulma-clean-theme
-theme: bulma-clean-theme
```
I'm not sure why you had this in here but it was stopping me from running it locally... My gemfile has the "github-pages" gem included which I think allows it to use the remote_theme...

eheinrich replied:
I had had the theme installed locally, but you're right, definitely better to just have the github-pages gem installed and use remote_theme

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+# Components
+
+The Observation Control System is composed of several python libraries, applications, and django web applications that work together to provide a complete upper stack solution to an observatory. The diagram below shows the major OCS components and how they are interconnected.
```
"The Observatory Control System (OCS) is"

"provide a complete upper stack solution to {control or manage} an observatory"

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+The Observation Control System is composed of several python libraries, applications, and django web applications that work together to provide a complete upper stack solution to an observatory. The diagram below shows the major OCS components and how they are interconnected.
+
+![OCS Components](/assets/images/ocs_applications.png)
```
Some notes on this diagram:
- It looks like there are two arrows really close together going to the Elasticsearch box
- Spelling in downtime box: "Downtime Databse"
- Can you center the text in the boxes?
- Maybe also add an arrow connecting users to the science archive?
- Potentially change text in scheduler box to use "observation requests" instead of "requests"

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+The core of the OCS is the [Observation Portal]({{ site.baseurl }}{% link components/observation_portal.md %}). It manages User accounts for all the other applications that need authentication, and it provides a database and API for submitting, updating, and viewing Proposals, Requests, and Observations.
+
+The [Configuration Database (Configdb)]({{ site.baseurl }}{% link components/configuration_database.md %}) manages the state of the observatory, as viewed by the other upper stack applications. It provides hierarchical models of the observatory's structure, from the site/enclosure/telescope down to the modes and properties of cameras within an instrument. The purpose of this information is to all ...
```
Suggest "is to allow automatic validation and auto-complete of observatory details when possible in other applications"

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+## Optional Components
+
+These applications all add on to the core OCS applications in various ways, but their inclusion into the software stack is optional. If an observatory already has a custom solution for one of the components, like scheduling or archiving of data, then those can be integrated with the Observation Portal rather than using the OCS [Adaptive Scheduler]({{ site.baseurl }}{% link components/adaptive_sch ...
```
Suggestion for first sentence: "The optional components are not required, but they all expand the capabilities of an OCS software stack in various ways."

Suggest "...of the components, such as scheduling or data archiving, then..."

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+The [Adaptive Scheduler]({{ site.baseurl }}{% link components/adaptive_scheduler.md %}) provides a python application which interfaces with the other OCS databases to turn a set of input Requests into a set of scheduled Observations in the Observation Portal. While the scheduler is fully integrated into the OCS ecosystem, it is possible to use another scheduling method instead.
+
+The [Science Archive application and Ingester library]({{ site.baseurl }}{% link components/science_archive.md %}) provide the means to upload data products into Amazon Web Services (AWS) S3 buckets, and store their metadata in a database to allow for easy searching, filtering, and bulk downloading of the data products by users. AWS S3 provides a relatively cheap and scalable way for observatorie ...
```
Suggest "...bulk downloading of data products by users using the Science Archive API."

Suggest "...AWS S3 provides a relatively inexpensive and scalable way for observatories to store data products redundantly and long-term..."

eheinrich commented on components/index.md:
```diff
@@ -1,8 +1,31 @@
+
+The [Science Archive application and Ingester library]({{ site.baseurl }}{% link components/science_archive.md %}) provide the means to upload data products into Amazon Web Services (AWS) S3 buckets, and store their metadata in a database to allow for easy searching, filtering, and bulk downloading of the data products by users. AWS S3 provides a relatively cheap and scalable way for observatorie ...
+
+The [Downtime Database]({{ site.baseurl }}{% link components/downtime_database.md %}) is a simple database, REST API, and web frontend for creating and retrieving periods of unschedulable time on a telescope or instrument_type. Using the downtime database isn't strictly necessary, but it can increase scheduling efficiency at the observatory since observations can be scheduled exactly around the b ...
```
Suggestion for first sentence "The Downtime Database is a simple database with a REST API and web frontend for creating and retrieving periods of planned downtime on a telescope or instrument."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+# Observation Portal
+
+The backbone of the Observatory Control System. This django application manages Users, Proposals, Requests, and Observations. The Observation Portal is deployed as a pypi library called django_ocs_observation_portal, and contains django apps for accounts, requests, observations, proposals, and sciapplications.
```
Suggest: "The Observation Portal is a Django application that manages Users, Proposals, Observation Requests, and Observations, and is the backbone of the Observatory Control System. It is available on PyPI as [django_ocs_observation_portal](https://pypi.org/project/django-ocs-observation-portal/), and contains Django apps for..."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Accounts
+
+This app provides the User accounts that are used for authentication between all OCS applications, using [oauth2](https://django-oauth-toolkit.readthedocs.io/en/latest/). It also provides a Profile model which describes account details.
```
"...Profile model which includes additional account details...."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Requests
+
+The OCS Request language is designed to be able to support a broad range of observation requests. The request models in the diagram below show the hierchary of properties of a RequestGroup.
```
"The models in the diagram..."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+![Request Models](/assets/images/request_models.png)
+
+- **RequestGroup** - The top level of the model. It contains one or more Requests and acts as a logical grouping to help the end user keep track of a set of Requests
+- **Request** - A single independent set of observations to be scheduled in one contiguous block on a resource
```
Maybe use something like "observing parameters" instead of "observations" since observations are a different model?

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+- **Constraints** - A set of observing constraints that must be met to schedule the observation
+- **Target** - The parameters describing the target, supporting ICRS, orbital elements, hour angle, or satellite coordinates.
+
+More information about the specific fields expected in each section of the RequestGroup can be found in the [api specification]({{ site.baseurl }}{% link api/downtime.md %}). Most fields have their values validated automatically from the acceptable values defined in the [Configuration Database (Configdb)]({{ site.baseurl }}{% link components/configuration_database.md %}). There are several areas  ...
```
api -> API

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+- **Guiding** - Parameters for the guiding of the observation
+- **Acquisition** - Parameters for the acquisition of the target
+- **Constraints** - A set of observing constraints that must be met to schedule the observation
+- **Target** - The parameters describing the target, supporting ICRS, orbital elements, hour angle, or satellite coordinates.
```
Remove the period at the end for consistency, the others don't have them

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+| ---------------- | ------------------ |
+| NORMAL | a normal Request whose priority is just the proposal's base priority multiplied by the `ipp_value` |
+| TIME_CRITICAL | base proposal priority is multiplied by 100. This factor is meant to beat out any NORMAL request, but all TIME_CRITICAL requests of the same priority are treated equally. This type is meant for observations that are coordinated between multiple observatories, or just have an extremely high priority |
+| RAPID_RESPONSE | These requests bypass normally scheduling and have their own scheduling loop that can pre-empt any non-RAPID_RESPONSE observation that is currently in progress. All requests of this type with the same base proposal priority are treated equally. These observation_types are very disruptive to the schedule, so they should only be used for observations that truely must begin immedi ...
```
"These requests bypass normal scheduling..."

truely -> truly

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+#### Extra Params
+
+Most of the levels of a RequestGroup have a json field called `extra_params` which accepts arbitrary key/value pairs that will be passed along to the telescope control software. This allows for complex non-standard configuration, instrument, target, or guiding and acquisition parameters to be added and used without modifying the Observation Portal code. Data passed through the `extra_params` fiel ...
```
json -> JSON

"in a handle of special cases" is "handle" supposed to be "couple" or "few"?

jnation3406 replied:
it was supposed to be "handful", which is probably similar to a couple or few...

eheinrich replied:
Ah makes sense. That would work too!

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Observations
+
+The observations app deals with scheduled or directly submitted observations. An Observation is scheduled Request, or a Request plus information about status, where the observation is scheduled, and when. The hierarchy of an Observation is shown below, but it is very similar to that of the Request except with a few additional pieces of information tacked on.
```
"An Observation is a scheduled Request"

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+![Observation Models](/assets/images/observation_models.png)
+
+- **Observation** - The top level of a Scheduled Request, this contains the *where* and *when* information,
+as well as some of the RequestGroup fields that the telescope control software might want to include in observation's data products
```
"...want to include in the observation's data products..."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+as well as some of the RequestGroup fields that the telescope control software might want to include in observation's data products
+- **Configuration Status** - This structure contains status information about a single Configuration of the Request.
+Observatory software is expected to report back status per Configuration in real time as it is attempted and completed.
+The contents of this structure will be merged into it's corresponding Configuration when retrieved through the API.
```
it's -> its

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+A RequestGroup can be submitted with one of three observation types, described in the table below. Each observation type has its own pool of allocated time to draw from, so users must be allocated time in the observation types they want to submit with.
+
+| observation_type | description |
```
I think it might be good to add "DIRECT" here as well, so that it can be described here what a directly submitted observation is

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Proposals
+
+Proposals are an essential part of the Observation Portal for authenticating who is allowed to submit observational Requests and on what resource. Proposals can have one or more Principle Investigators (PIs) and any number of Co-Investigators (COIs) assigned to them. They can have any number of Time Allocations associated with them for any number of observing Semesters. The book keeping for the d ...
```
"...authenticating who is allowed to submit observation requests and on what..."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+Proposals are an essential part of the Observation Portal for authenticating who is allowed to submit observational Requests and on what resource. Proposals can have one or more Principle Investigators (PIs) and any number of Co-Investigators (COIs) assigned to them. They can have any number of Time Allocations associated with them for any number of observing Semesters. The book keeping for the d ...
+
+Proposals are currently added to the system in one of two ways - either through the admin interface, or by porting them over via the Science Application process in the sciapplications app.
```
"...either through the Django admin interface, or by generating them via..."

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Sciapplications
+
+This application handles the proposal submission and review process. Calls for applications can be created via the admin interface, and eligible users can submit Science Applications via the API for consideration. After review, the accepted science applications can be automatically converted into Proposals with time allocations for the coming observing semester. This application can be completely ...
```
"This Django app handles..." or something using "app" instead of "application" for consistency with the other app descriptions

"Calls for science applications can..."

consistently capitalize or lowercase "science applications"

Observatory's -> observatories

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+## How to Customize
+
+The first level of customization comes from the instrument and observatory definations within Configuration Database. By using Generic Modes, Optical Path Elements, and custom validation_schema, a lot of unique validation rules can be added to the Observation Portal without modifying any code. For more advanced customization, the [Observation Portal Project](https://github.com/observatorycontrols ...
```
definations -> definitions

overwriting and overwritting -> overriding

"...methods of Models which are used for generating..."

jnation3406 replied:
wow that's embarrassing, I need to get a vscode spellchecker plugin

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### Serializer Overrides
+
+The serializers define the validation logic for all the models of the Observation Portal. Any serializer can be overwritten by defining a serializer class that subclasses the corresponding library serializer and overrides any serializer method - usually the `validate` method. Here is an example serializer override that enforces that spectrograph instruments must include calibrations with their sp ...
```
overwritten -> overridden

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+SPECTRO_CATEGORY = 'SPECTRA'
+
+class MyConfigurationSerializer(RequestSerializer):
```
Maybe call this `MyRequestSerializer` instead?

eheinrich commented on components/observation_portal.md:
```diff
@@ -0,0 +1,118 @@
+
+### as_dict Overrides
+
+We use `as_dict` methods within the django Models to efficiently serialize our return values for API calls. This circumvents the normal serialization behaviour within the django rest framework serializers, but it greatly speeds up the call, which is important when you have a large number of Requests. Overriding the as_dict method allows you to add fields or non-model data to any API response. The ...
```
django -> Django (at least, I think it should be capitalized? that's what django does by convention at least https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/#commonly-used-terms)

"...efficiently serialize return values for API calls..."

overwritten -> overridden

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+hide_hero: true
+---
+
+# Configuration Databse (configdb)
```
Databse -> Database

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+# Configuration Databse (configdb)
+
+This django application manages details about the observatory's Sites, Enclosures, Telescopes, Instruments, and Cameras. It stores a hierarchical representation of what is available at the observatory, which is used for automatic validation within the other OCS applications. It also contains timing information for pieces of observations with instruments, which is used in the Observation Portal to ...
```
Suggest: "It also contains timing information, storing overheads for different parts of an observation, which is used in the Observation Portal to..."

jnation3406 replied:
changed this a bit more to "It also contains overheads for different parts of an observation for each instrument, which is used in the Observation Portal to..."

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+## Structure
+
+![Configdb Models](/assets/images/configdb_models.png)
```
Comment on the diagram - Instead of Observatory, use Enclosure

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+![Configdb Models](/assets/images/configdb_models.png)
+
+- **Site** - The top level of the model. It contains a 3 digit site code (often the closest airport code), and lat/lon/timezone information
```
"3 letter" or "3 character" instead of "3 digit"

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+![Configdb Models](/assets/images/configdb_models.png)
+
+- **Site** - The top level of the model. It contains a 3 digit site code (often the closest airport code), and lat/lon/timezone information
+- **Enclosure** - A 4 digit enclosure code representing a physical enclosure at a site
```
"letter" or "character" instead of "digit"

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+- **Site** - The top level of the model. It contains a 3 digit site code (often the closest airport code), and lat/lon/timezone information
+- **Enclosure** - A 4 digit enclosure code representing a physical enclosure at a site
+- **Telescope** - Contains a 4 digit telescope code as well as telescope specific information that aids in calculating target visibility, such as the horizon and hour angle limits and zenith blind spot of the telescope
```
"letter" or "character" instead of "digit"

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+- **Camera** - Specific instance of a **Camera Type** with set of **Optical Elements Groups** associated with it
+- **Camera Type** - The details of a class of cameras, such as their pixel scale and resolution
+- **Optical Element Group** - A grouping of **Optical Element**s of the same type with an optional default and an element change overhead
+- **Optical Element** - A generic code for a single optical path element that can be allowed for scheduling or not. This can encapsulate anything in the optical path of the instrument, including filters, slits, diffusers, or gratings
```
Suggest: "that can be made available for scheduling or not."

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+## Special features and Generics
+
+In this section we will highlight a few of the generic aspects of the Configuration Databse, which can allow users to customize properties to fit their instruments better
```
Add period

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+### Generic Modes
+
+The Generic Modes and Generic Mode Groupings allow for defining arbitrary *types* of mode sets, which can assist in custom validation within the Observation Portal by overriding corresponding serializers. The default Generic Mode types used by the base Observation Portal listed below, but additional types can be added as needed.
```
Missing an "are": "The default Generic Mode types used by the base Observation Portal are listed below"

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+| Type | Description |
+| ---- | ----------- |
+| readout | Validates against the **Instrument Config** `mode` attribute in a **Request**. A named readout mode should be defined and interpretted by the telescope control software to encapsulate any number of instrument parameters, including things like binning |
```
interpretted -> interpreted

jnation3406 replied:
at least i was consistently wrong on this one

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+| Type | Description |
+| ---- | ----------- |
+| readout | Validates against the **Instrument Config** `mode` attribute in a **Request**. A named readout mode should be defined and interpretted by the telescope control software to encapsulate any number of instrument parameters, including things like binning |
+| acquisition | Validates against the **Acquisition Config** `mode` in a **Request**. A named acquisition mode should be defined and interpretted by the telscope control software to encapsulte different types of acquisition. If additional properties need to be user specified, they can be required within the validation schema of the mode |
```
interpretted -> interpreted

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+| ---- | ----------- |
+| readout | Validates against the **Instrument Config** `mode` attribute in a **Request**. A named readout mode should be defined and interpretted by the telescope control software to encapsulate any number of instrument parameters, including things like binning |
+| acquisition | Validates against the **Acquisition Config** `mode` in a **Request**. A named acquisition mode should be defined and interpretted by the telscope control software to encapsulte different types of acquisition. If additional properties need to be user specified, they can be required within the validation schema of the mode |
+| guiding | Validates against the **Guiding Config** `mode` in a **Request**. A named guiding mode should be defined and interpretted by the telscope control software to encapsulte different methods of guiding. If additional properties need to be user specified, they can be required within the validation schema of the mode |
```
interpretted -> interpreted

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+| readout | Validates against the **Instrument Config** `mode` attribute in a **Request**. A named readout mode should be defined and interpretted by the telescope control software to encapsulate any number of instrument parameters, including things like binning |
+| acquisition | Validates against the **Acquisition Config** `mode` in a **Request**. A named acquisition mode should be defined and interpretted by the telscope control software to encapsulte different types of acquisition. If additional properties need to be user specified, they can be required within the validation schema of the mode |
+| guiding | Validates against the **Guiding Config** `mode` in a **Request**. A named guiding mode should be defined and interpretted by the telscope control software to encapsulte different methods of guiding. If additional properties need to be user specified, they can be required within the validation schema of the mode |
+| rotator | Validates against the **Instrument Config** `rotator_mode` in a **Request**. A named acquisition mode should be defined and interpretted by the telscope control software to encapsulte different types of acquisition. If additional properties need to be user specified, they can be required within the validation schema of the mode |
```
interpretted -> interpreted

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+### Optical Elements
+
+Much like the Generic Mode Groups, the Optical Element Groups allow for defining arbitrary *types* of sets of optical elements for an **Camera**. The optical element groups have a element change overhead that is factored into Request durations, and the elements themselves can be marked as schedulable or not. The intention of generic Optical Elements is to support anything user configurable that i ...
```
an -> a in "allow for defining arbitrary types of sets of optical elements for an Camera"
a -> an in "The optical element groups have a element change"

eheinrich commented on components/configuration_database.md:
```diff
@@ -0,0 +1,72 @@
+
+### Cerberus Validation Schema
+
+The **Instrument Type** and **Generic Mode** models currently contain a field called `validation_schema`, which accepts a [Cerberus validation schema](https://docs.python-cerberus.org/en/stable/schemas.html) as a json blob. This validation schema is applied to the section of the **Request** in which the mode or instrument_type applies. Validation schemas can be used to enfore that existing fields ...
```
json -> JSON

eheinrich commented on components/adaptive_scheduler.md:
```diff
@@ -0,0 +1,36 @@
+
+# Adaptive Scheduler
+
+A python script that continuously runs the scheduling loop, optimizing the effective priority over the scheduling horizon. It utilizes a MIPS solver to optimize for the largest effective priority on the telescope network over your observing horizon. It uses [Googles OR Tools](https://github.com/google/or-tools) for the solver, which supports multiple free or comercial optimization algorithms. It  ...
```
comercial -> commercial

Add a period to the last sentence

eheinrich commented on components/adaptive_scheduler.md:
```diff
@@ -0,0 +1,36 @@
+
+## How it works
+
+The scheduler gets data from the other OCS apps to know what, when, and where it can schedule. A diagram showing the flow of a scheduling cycle and its communication with the other apps is shown below.
```
Suggest apps -> components in "...from the other OCS components to know what, when, and where it can schedule observations. A diagram showing the flow of a scheduling cycle and its communication with the other components is shown below."

eheinrich commented on components/adaptive_scheduler.md:
```diff
@@ -0,0 +1,36 @@
+
+![Scheduler Workflow](/assets/images/scheduler_workflow.png)
+
+The ConfigDB and Observation Portal connections are essential, but the DowntimeDB and TelemetryDB connections are both optional. The scheduler starts by getting the set of schedulable telescopes, by looking at what sites and telescopes are `active` in ConfigDB and optionally by looking at an Elasticsearch DB for weather and availability telescope telemetry. It then gets the set of schedulable Req ...
```
"weather and availability telescope telemetry" -> "weather and telescope availability telemetry"

eheinrich commented on components/adaptive_scheduler.md:
```diff
@@ -0,0 +1,36 @@
+
+### Telemetry
+
+The TelemetryDB is currently expecting weather and telescope availability telemetry in a specific format within an ElasticSearch database, but the code could be extended to support other data sources and formats for weather data. The format of data it currently expects is referenced in the [code here](https://github.com/observatorycontrolsystem/adaptive_scheduler/blob/main/adaptive_scheduler/moni ...
```
Suggest "The TelemetryDB currently expects weather and telescope availability telemetry in a specific format within an ElasticSearch database, but the code can be extended to support..."

eheinrich commented on components/science_archive.md:
```diff
@@ -0,0 +1,17 @@
+
+# Science Archive
+
+This django application manages manages the observatory's data products, and references the raw data products which are stored in AWS S3. It features a REST API for querying for the metadata and download links for data products, and for submitting the metadata and S3 location of newly ingested data products. For the complete list of filterable fits header values, check out the [API specification] ...
```
"manages manages" -> "manages"

Suggest "It features a REST API for retrieving metadata and download links for data products, and also for submitting metadata and data products for upload to S3."

fits -> FITS

eheinrich commented on components/science_archive.md:
```diff
@@ -0,0 +1,17 @@
+
+## Ingester Library
+
+This python library `ocs_ingester` is used to upload data products to an AWS S3 bucket, and to update the **Science Archive** with metadata for that data product. It validates that the metadata contains all the required values, and adds in values for which it can determine a default value. It can accept data products of any file type, but if it is not a .fits file, the proper fits header data wil ...
```
Suggest "This python library is available on PyPI as`ocs_ingester`, and is..."
.fits -> FITS

eheinrich commented on components/downtime_database.md:
```diff
@@ -0,0 +1,16 @@
+
+# Downtime Database
+
+This django application manages a REST API and web frontend for creating and retrieving periods of downtime at a telescope or instrument_type. Downtime periods are time ranges in which scheduling is blocked in advance for any number of reasons, including maintenance, non-robotic usage, or certain instruments being offline.
```
Suggest "...web frontend for creating and retrieving periods of downtime on a telescope or instrument type"

eheinrich commented on components/downtime_database.md:
```diff
@@ -0,0 +1,16 @@
+
+This django application manages a REST API and web frontend for creating and retrieving periods of downtime at a telescope or instrument_type. Downtime periods are time ranges in which scheduling is blocked in advance for any number of reasons, including maintenance, non-robotic usage, or certain instruments being offline.
+
+This application is optional in the OCS ecosystem. It is useful to block out downtimes in advance for the scheduler to be able to scheduler observations around them, and for the downtimes to not interrupt any ongoing observations. If that is not a big concern, then simply switching the telescopes to inactive in configdb or changing the state of instruments to be unschedulable will prevent those r ...
```
Suggest "...downtimes in advance so that the scheduler can schedule observations around those periods of downtime, which can help prevent interrupts to ongoing observations."

eheinrich reviewed (commented):
This is great! And must have taken quite a lot of work. Most of the comments that I left are for minor typos and sentence refactors.

One thing I wanted to decided on is that I've been using the term "observation requests" instead of "requests" for writing docs because for the uninitiated, using just the word "requests" might not be descriptive enough, and also saying that observation requests are turned into scheduled observations, or into an observing schedule I think is more clear than saying requests are turned into observations. This also helps a bit cause sometimes we use the word "request" to mean an HTTP request... of course it doesn't track exactly with the name of the model but we hardly ever say "RequestGroup" anyway and for high level docs, it probably doesn't matter as much. In any case, I just want to make sure we're consistent in the docs so I can update the things I've written if you disagree. What do you think?

Commit:
Addressed review comments and udpated diagrams

jnation3406 commented:
I addressed all the review comments, but I'm not sure what to do for two conventions. First is what you described with the requests = "observing requests" and observations = "scheduled observations". That sounds okay to use in some places, but in places where we are discussing both observations and requests, I think it muddies it up a bit because you use "observing/observation" in the same sentence too much. 

The other thing I have a question about is capitalization of things like django apps or models. In a lot of places I tried to capitalize those things for emphasis, and so the reader could relate those back to previous sections or definitions of those things (i.e. try to make them look like proper nouns vs the generic word). So there is a lot of Requests and Proposals and stuff in there, but I'm not sure if you want all of those lower-case throughout, especially if we are changing all Requests to Observing Requests?

Commit:
Normalized capitalization and added some more links

eheinrich committed:
Merge branch 'feature/general-docs' into add_components
